server/service/src/sync/sync_user.rs

```diff
@@ -0,0 +1,85 @@
+use chrono::NaiveDateTime;
+use log::info;
+use repository::{RepositoryError, UserAccountRow, UserAccountRowRepository};
+
+use crate::{
+    auth_data::AuthData,
+    login::{FetchUserError, LoginError, LoginInput, LoginService},
+    service_provider::ServiceProvider,
+};
+
+pub struct SyncUser {}
+
+impl SyncUser {
+    pub async fn update_user(
+        service_provider: &ServiceProvider,
+        auth_data: &AuthData,
+        user_id: &str,
+    ) -> Result<UserAccountRow, LoginError> {
+        let ctx = service_provider.basic_context()?;
+
+        let central_server_url = service_provider
+            .settings
+            .sync_settings(&ctx)?
+            .ok_or(LoginError::InternalError(
+                "Cannot find sync settings".to_string(),
+            ))?
+            .url;
+        let username = UserAccountRowRepository::new(&ctx.connection)
+            .find_one_by_id(user_id)?
+            .ok_or(LoginError::FetchUserError(FetchUserError::InternalError(
+                "Could not find user".to_string(),
+            )))?
+            .username;
+        let password = get_password(&auth_data, &user_id);
+
+        match LoginService::fetch_user_from_central(&LoginInput {
```

Comment:
Fetching user and updating instead of using the login function since not updating token